lib/consumer/amqp_data_consumer.ex

```diff
@@ -16,7 +16,7 @@ defmodule Mississippi.Consumer.AMQPDataConsumer do
 
   # TODO should this be customizable?
   @reconnect_interval 1_000
-  @connection if Mix.env() == :test, do: MockAMQPConnection, else: ExRabbitPoolConnection
+  @connection Application.compile_env(:mississippi, [:consumer, :amqp_data_consumer, :connection], ExRabbitPoolConnection)
```

Comment:
Note that Mississippi (neither Consumer nor Producer) is not an `Application`, so reading this from the `Application` environment would imply that we're polluting the config of the application depending on us. See https://hexdocs.pm/elixir/main/design-anti-patterns.html#using-application-configuration-for-libraries

Reply:
It is an application as defined from the mix.exs. If it weren't and the `:mississippi` application didn't exist, that line would error out. If you `s/mississippi/non_existing_app` in `config/test.exs` and the `@connection` declaration: `You have configured application :non_existing_app in your configuration file, but the application is not available.` ).
With that said, it is true that it's better to avoid using it eg if a project wants to use two instances of mississippi, so I've opted for setting a start option instead